### PR TITLE
[21.05] Ceph Nautilus

### PR DESCRIPTION
**Pull request by osnyx**
@flyingcircusio/release-managers

This adds support in our platform for the Ceph Nautilus release, adapting tooling (fc-ceph) and connected roles (kvm, backy) accordingly.
The Jewel and Luminous releases are still kept around for compatibility and continue to function as before.

- [ ] the KVM role is not ready for Ceph Nautilus yet and needs to be kept at Luminous for now

## Release process

Impact: internal version upgrade for our storage cluster

Changelog:

- Ceph software upgrade to Nautilus
- structured config for Ceph roles and services
  - backwards incompatibility: mixing of structured config and old config lines not possible
- Nautilus provides a thinner ceph-client package for usage on non-server hosts

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined?
  - [x] software upgrade, potentially fixes upstream bugs
  - [x] new release breaking changes need to be dealt with by reviewing release notes
  - [x] tests must still pass, no new regressions introduced
- [x] Security requirements tested? (EVIDENCE)
  - [x] reviewed release notes and upgrade notes
  - [x] operability checks in development cluster
  - [x] automated tests pass

### CODE REVIEW

**Comment by osnyx** [comment]
related PRs:
https://github.com/flyingcircusio/backy/pull/23
a fc.qemu PR will follow

File: pkgs/fc/ceph/src/fc/ceph/api/tests/fixtures/ceph.conf [hunk 1]
```diff
@@ -28,7 +28,9 @@ rbd cache target dirty = 33554432
 rbd default format = 2
 
 [mon]
-mon addr = 192.168.23.10:6789,192.168.23.4:6789,192.168.23.9:6789
+# FIXME: mon addr is explicitly deprecated from Nautilus on, let's see whether a public
```
Comment (ctheune) [inline pkgs/fc/ceph/src/fc/ceph/api/tests/fixtures/ceph.conf]: according to @osnyx this has been fixed

File: pkgs/fc/default.nix [hunk 2]
```diff
@@ -9,12 +9,28 @@ rec {
   check-ceph = {
     jewel = callPackage ./check-ceph/jewel {ceph = pkgs.ceph-jewel;};
     luminous = callPackage ./check-ceph/luminous {ceph = pkgs.ceph-luminous;};
+    nautilus = callPackage ./check-ceph/luminous {ceph = pkgs.ceph-nautilus.ceph-client;};
```
Comment (ctheune) [inline pkgs/fc/default.nix:13]: comment would be nice

File: pkgs/fc/default.nix [hunk 3]
```diff
@@ -35,12 +51,26 @@ rec {
...
+    libceph = pkgs.ceph-nautilus.libceph;
+  };
 
   secure-erase = callPackage ./secure-erase {};
   sensuplugins = callPackage ./sensuplugins {};
   sensusyntax = callPackage ./sensusyntax {};
   userscan = callPackage ./userscan.nix {};
-  util-physical = callPackage ./util-physical {};
+  util-physical = {
+    jewel = callPackage ./util-physical/ceph-luminous {ceph = pkgs.ceph-jewel;};
```
Comment (ctheune) [inline pkgs/fc/default.nix:73]: comment

**Review by ctheune — CHANGES_REQUESTED** [review]
minor changes

**Pull request merged**